gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/README.adoc

```diff
@@ -0,0 +1,247 @@
+= Gravitee.io API Management - Gateway Testing SDK
+
+ifdef::env-github[]
+image:https://ci.gravitee.io/buildStatus/icon?job=gravitee-io/gravitee-gateway/master["Build status", link="https://ci.gravitee.io/job/gravitee-io/job/gravitee-gateway"]
+image:https://f.hubspotusercontent40.net/hubfs/7600448/gravitee-github-button.jpg["Join the community forum", link="https://community.gravitee.io?utm_source=readme", height=20]
+endif::[]
+
+== Overview
+
+The Gateway Testing SDK is a library used to test your policy implementation in real condition.
+
+It provides pieces of code allowing you to easily run a gateway with your policy deployed for testing purpose.
+Then, you just have to write tests as usual.
+
+== A simple example
+
+Here is a test for the https://github.com/gravitee-io/gravitee-policy-mock[Mock policy].
+
+```java
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.gravitee.apim.gateway.tests.sdk.AbstractPolicyTest;
+import io.gravitee.apim.gateway.tests.sdk.annotations.DeployApi;
+import io.gravitee.apim.gateway.tests.sdk.annotations.GatewayTest;
+import io.gravitee.policy.mock.MockPolicy;
+import io.gravitee.policy.mock.configuration.MockPolicyConfiguration;
+import io.reactivex.observers.TestObserver;
+import io.vertx.reactivex.core.buffer.Buffer;
+import io.vertx.reactivex.ext.web.client.HttpResponse;
+import io.vertx.reactivex.ext.web.client.WebClient;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author GraviteeSource Team
+ */
+@GatewayTest
+@DeployApi("/apis/mock.json")
+class MockPolicyIntegrationTest extends AbstractPolicyTest<MockPolicy, MockPolicyConfiguration> {
+
+    @Test
+    @DisplayName("Should use mock without calling endpoint")
+    void shouldUseMock(WebClient client) {
+        wiremock.stubFor(get("/endpoint").willReturn(ok("response from backend")));
+
+        final TestObserver<HttpResponse<Buffer>> obs = client.get("/test").putHeader("reqHeader", "reqHeaderValue").rxSend().test();
+
+        awaitTerminalEvent(obs);
+        obs
+            .assertComplete()
```

Comment:
Didn't you change the SDK to make `awaitTerminalEvent` returning the `obs` ?
```suggestion
        awaitTerminalEvent(obs)
            .assertComplete()
```

Reply:
yes it's changed, but the code is still valid :p

Reply:
see [e5b99f6](https://github.com/gravitee-io/gravitee-api-management/pull/1939/commits/e5b99f670590cbfb3e8d811e22901ee754869561)